### PR TITLE
Set O_TRUNC and BIG_WRITES support in jnifuse

### PR DESCRIPTION
**Pull request by LuQQiu**
partial fix for https://github.com/Alluxio/alluxio/issues/14895
### What changes are proposed in this pull request?

Set O_TRUNC and BIG_WRITES support in jnifuse

### Why are the changes needed?

Set if the filesystem supports the O_TRUNC open flag.
Set if the filesystem can handle writes larger than 4 KB.

### Does this PR introduce any user facing changes?

Author: [liuyongqing](https://github.com/liuyongqing)

### CODE REVIEW

**Comment by LuQQiu** [comment]
@maobaolong PTAL, thanks

File: integration/jnifuse/native/src/main/native/libjnifuse/jnifuse_impls.cc [hunk 1]
```diff
@@ -14,6 +14,19 @@
...
+  #ifndef __APPLE__
+  if((unsigned int)conn->capable & FUSE_CAP_ATOMIC_O_TRUNC){
+      conn->want |= FUSE_CAP_ATOMIC_O_TRUNC;
+  }
+  #endif
+
+  if((unsigned int)conn->capable & FUSE_CAP_BIG_WRITES){
+    conn->want |= FUSE_CAP_BIG_WRITES;
+  }
+  return NULL;
```
Comment (yyongycy) [inline integration/jnifuse/native/src/main/native/libjnifuse/jnifuse_impls.cc:27]: always return a NULL?

**Comment by maobaolong** [comment]
@LuQQiu Would you like to add init callback into java side also?

**Comment by LuQQiu** [comment]
but we do need to improve Alluxio side parameter checking

File: integration/jnifuse/native/src/main/native/libjnifuse/jnifuse_impls.cc (hunk 1 above)
Comment (yyongycy) [reply integration/jnifuse/native/src/main/native/libjnifuse/jnifuse_impls.cc:27]: https://libfuse.github.io/doxygen/structfuse__operations.html#a3a1f9562a38871caaf4fc2de7ea5b470

It should be fine as long as it is only called once.

**Review by yyongycy — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
Error: 3.615 [ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 56.073 s <<< FAILURE! - in alluxio.client.fs.concurrent.ConcurrentFileInStreamIntegrationTest
Error: 3.622 [ERROR] alluxio.client.fs.concurrent.ConcurrentFileInStreamIntegrationTest.FileInStreamConcurrency  Time elapsed: 26.069 s  <<< FAILURE!
java.lang.AssertionError: Failed with except

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**